pkg/build/controller/build/build_controller.go

```diff
@@ -1373,6 +1373,7 @@ func setBuildCompletionData(build *buildapi.Build, pod *v1.Pod, update *buildUpd
 			update.setLogSnippet(msg)
 		}
 	}
+	metrics.BuildTerminated.WithLabelValues(string(build.Status.Phase), string(build.Status.Reason), buildapi.StrategyType(build.Spec.Strategy)).Inc()
```

Comment:
setbuildcompletiondata is going to get called for every terminated build on every relist.  So you're going to count terminated builds repeatedly.

The only safe way to do this is to count the build when the controller puts it in the terminal state, and since it's not always the controller putting it in the terminal state (sometimes the pod puts the build in the terminal state) you really can't do this effectively.

if we're sure that "build.Status.CompletionTimestamp" is not set by the build pod, you could consider putting this line inside that if block, but you'd have to be absolutely certain we were guaranteed to go through that code path exactly one time for every build, when it reaches a terminal state.